pkg/kubelet/kubelet.go

```diff
@@ -694,6 +695,42 @@ const (
 	NetworkContainerImage = "kubernetes/pause:latest"
 )
 
+func (kl *Kubelet) pullAndRunContainer(pod *api.BoundPod, container *api.Container, podVolumes volumeMap, netMode string) (id dockertools.DockerID, err error) {
+	ref, err := containerRef(pod, container)
+	if err != nil {
+		glog.Errorf("Couldn't make a ref to pod %v, container %v: '%v'", pod.Name, container.Name, err)
+	}
+
+	kl.pullLock.RLock()
```

Comment:
I don't think that we need to lock around container start, please see where I pushed the lock to @ head.